### PR TITLE
[all] ADD Doxygen comments on Datas

### PR DESCRIPTION
**Pull request by guparan**
This work aims at having a complete Doxygen documentation of SOFA components, especially concerning their Datas.

I used the shell script doxygenDataComments.sh to
1. Force each Data to be defined on one line
2. Add an inlined and Doxygen-formatted comment on each Data (if no comment found) copying the description used in `initData`

Obviously this has not a 100% coverage (it "only" affects Data<.*> written members) but it's still a very good start.

PS: our nightly-generated Doxygen is ready to receive these changes: https://www.sofa-framework.org/newapi

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
I like the idea of using the comment in initData.

File: applications/plugins/SofaImplicitField/components/geometry/DiscreteGridField.h [hunk 1]
```diff
@@ -71,8 +71,10 @@ class SOFA_SOFAIMPLICITFIELD_API DiscreteGridField : public virtual ScalarField
     int getNextDomain();
 
     sofa::core::objectmodel::DataFileName d_distanceMapHeader;
-    Data< int > d_maxDomains;
-    Data< double > dx, dy, dz;    // translation of original image
+    Data< int > d_maxDomains; ///< Number of domains available for caching
+    Data< double > dx;    // translation of original image
+    Data< double > dy;    // translation of original image
+    Data< double > dz;    // translation of original image
```
Comment (guparan) [inline applications/plugins/SofaImplicitField/components/geometry/DiscreteGridField.h]: This should be Doxygen comments, will do another commit

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

**Comment by guparan** [comment]
Please consider merging this PR soon ;-)
[ci-build][with-scene-tests]

**Comment by damienmarchal** [comment]
So do I ?

**Comment by guparan** [comment]
Yay \o/

**Comment by damienmarchal** [comment]
Même pas peur

**Comment by guparan** [comment]
This is gonna be HUGE 👍

**Comment by damienmarchal** [comment]
It is not strictly equivalent but at Defrost we are considering to automatically bind the sofa components to python classes.

This is done in:  
https://github.com/SofaDefrost/SofaComponents

The resulting side effect is that we have an auto-generated documentation of components and their data at:
http://sofacomponents.readthedocs.io/en/latest/
